src/python/pants/engine/process_test.py

```diff
@@ -107,6 +112,62 @@ def test_failing_process(rule_runner: RuleRunner) -> None:
     assert "Process 'failure' failed with exit code 1." in str(exc.value)
 
 
```

Comment:
These tests demonstrate the semantics, and provide critical coverage: they're probably the most important thing to scrutinize in here.